pkg/commands/venv.go

```diff
@@ -1,16 +1,168 @@
+		Environment:            cluster.Environment,
+		Repository:             meta.Name,
+		ClusterName:            cluster.Name,
+		UserDataDir:            userDataDir,
+		Debug:                  o.Debug,
+		KubectlBinaryDir:       path.Dir(k.BinaryPath),
+		AwsIamAuthenticatorDir: path.Dir(a.BinaryPath),
+	}
+
+	err = opts.Validate()
+	if err != nil {
+		return OkctlEnvironment{}, errors.E(err, "failed to validate show credentials options")
+	}
+
+	return opts, nil
+}
+
+// GetOkctlEnvVars converts an okctl environment to a map with environmental variables
+func GetOkctlEnvVars(opts OkctlEnvironment) map[string]string {
+	appDir := opts.UserDataDir
+
+	kubeConfig := path.Join(appDir, config.DefaultCredentialsDirName, opts.ClusterName, config.DefaultClusterKubeConfig)
+	awsConfig := path.Join(appDir, config.DefaultCredentialsDirName, opts.ClusterName, config.DefaultClusterAwsConfig)
+	awsCredentials := path.Join(appDir, config.DefaultCredentialsDirName, opts.ClusterName, config.DefaultClusterAwsCredentials)
+
+	h := &helm.Config{
+		HelmPluginsDirectory: path.Join(appDir, config.DefaultHelmBaseDir, config.DefaultHelmPluginsDirectory),
+		HelmRegistryConfig:   path.Join(appDir, config.DefaultHelmBaseDir, config.DefaultHelmRegistryConfig),
+		HelmRepositoryCache:  path.Join(appDir, config.DefaultHelmBaseDir, config.DefaultHelmRepositoryCache),
+		HelmRepositoryConfig: path.Join(appDir, config.DefaultHelmBaseDir, config.DefaultHelmRepositoryConfig),
+		HelmBaseDir:          path.Join(appDir, config.DefaultHelmBaseDir),
+		Debug:                opts.Debug,
+	}
+
+	envMap := make(map[string]string)
+
+	for k, v := range h.Envs() {
+		if k == "HOME" || k == "PATH" {
+			continue
+		}
+
+		envMap[k] = v
+	}
+
+	envMap["AWS_CONFIG_FILE"] = awsConfig
+	envMap["AWS_SHARED_CREDENTIALS_FILE"] = awsCredentials
+	envMap["AWS_PROFILE"] = "default"
+	envMap["KUBECONFIG"] = kubeConfig
+	envMap["PATH"] = getPathWithOkctlBinaries(opts)
+
+	return envMap
+}
+
+func getPathWithOkctlBinaries(opts OkctlEnvironment) string {
+	okctlPath := fmt.Sprintf("%s:%s", opts.KubectlBinaryDir, opts.AwsIamAuthenticatorDir)
+	osPath, osPathExists := os.LookupEnv("PATH")
+
+	if osPathExists {
+		return fmt.Sprintf("%s:%s", okctlPath, osPath)
+	}
+
+	return okctlPath
+}
+
+// MergeEnvVars first converts the given slice to a map. The provided slice must contain strings on the form "KEY=VALUE.
+// It then megres this map with the other provided map.
+// If both map contains a PATH key, they will be merged.
+func MergeEnvVars(osEnvs []string, venvMap map[string]string) map[string]string {
+	merged := make(map[string]string)
+
+	for key, val := range venvMap {
+		merged[key] = val
 	}
 
-	shell, ok = osLookupEnv("SHELL")
-	if ok {
-		return shell
+	osEnvMap := toMap(osEnvs)
+	for key, val := range osEnvMap {
+		merged[key] = val
+	}
+
+	// Merge PATHs
+	venvPath, venvHasPath := venvMap["PATH"]
+	osPath, osHasPath := osEnvMap["PATH"]
+
+	if osHasPath && venvHasPath {
```

Comment:
I'm just wondering in what situation would the venv PATH and the os PATH differ? Wouldn't they either be equal or the venv PATH empty?

Reply:
The venv PATH will always differ, because the venv path contains the path kubectl and aws-iam-authenticator. Also, unless OKCTL_NO_PS1 is set, the venv path will contain the path to the PS1 executable (venv_ps1).